wcomponents-core/src/main/java/com/github/bordertech/wcomponents/validation/AbstractWFieldIndicator.java

```diff
@@ -52,17 +52,19 @@ public AbstractWFieldIndicator(final WComponent relatedField) {
 	 * @param severity A Diagnostic severity code. e.g. {@link Diagnostic#ERROR}
 	 */
 	protected void showIndicatorsForComponent(final List<Diagnostic> diags, final int severity) {
-		FieldIndicatorModel model = getOrCreateComponentModel();
-		model.diagnostics.clear();
-		UIContext uic = UIContextHolder.getCurrent();
+		if (diags.size() > 0) {
```

Comment:
Can you be sure diags is not null at this point? Is null a valid value for the size arg or should you throw an illegal argument exception?
What if there is an existing model with diagnostics - should they be cleared if the new list of Diagnostics is null or empty? The are in the old code but in your adaptation an existing diagnostics list is only cleared if the new list has items.

Reply:
How about this at the beginning?
```
if (diags == null || diags.isEmpty()) {
	throw new IllegalArgumentException("Diagnostic list " + diags + "cannot be null or empty!");
}
```